tests/sur/routeVeeam.js

```diff
@@ -0,0 +1,582 @@
+    </SystemInfo>\n`;
+
+const testSystemMd5 = crypto.createHash('md5')
+    .update(testSystem, 'utf-8')
+    .digest('hex');
+
+const invalidTestSystem = `<?xmlversion="1.0"encoding="UTF-8"?>
+    <SystemInfo>
+       <ProtocolVersion>"1.0"</ProtocolVersion>
+       <ModelName>"ARTESCA"</ModelName>
+       <ProtocolCapabilities>
+          <CapacityInfo>true</CapacityInfo>
+          <UploadSessions>false</UploadSessions>
+          <IAMSTS>true</IAMSTS>
+       </ProtocolCapabilities>
+       <APIEndpoints>
+            <IAMEndpoint>a</IAMEndpoint>
+            <STSEndpoint>a</STSEndpoint>
+       </APIEndpoints>
+       <SystemRecommendations>
+           <S3ConcurrentTaskLimit>0</S3ConcurrentTaskLimit>
+           <S3MultiObjectDeleteLimit>-1</S3MultiObjectDeleteLimit>
+           <StorageCurrentTasksLimit>0</StorageCurrentTasksLimit>
+           <KbBlockSize>256</KbBlockSize>
+       </SystemRecommendations>
+    </SystemInfo>\n`;
+
+const invalidTestSystemMd5 = crypto.createHash('md5')
+    .update(testSystem, 'utf-8')
+    .digest('hex');
+
+let bucketUtil;
+let s3;
+
+/** makeVeeamRequest - utility function to generate a request going
+ * through veeam route
+ * @param {object} params - params for making request
+ * @param {string} params.method - request method
+ * @param {string} params.bucket - bucket name
+ * @param {string} params.objectKey - object key
+ * @param {object} [params.headers] - headers and their string values
+ * @param {object} [params.authCredentials] - authentication credentials
+ * @param {object} params.authCredentials.accessKey - access key
+ * @param {object} params.authCredentials.secretKey - secret key
+ * @param {string} [params.requestBody] - request body contents
+ * @param {object} [params.queryObj] - request query parameters
+ * @param {function} callback - with error and response parameters
+ * @return {undefined} - and call callback
+ */
+function makeVeeamRequest(params, callback) {
+    const { method, headers, bucket, objectKey,
+        authCredentials, requestBody, queryObj } = params;
+    const options = {
+        authCredentials,
+        hostname: ipAddress,
+        port: 8000,
+        method,
+        headers,
+        path: `/_/veeam/${bucket}/${objectKey}`,
+        urlForSignature: `/${bucket}/${objectKey}`,
+        requestBody,
+        jsonResponse: false,
+        queryObj,
+    };
+    makeRequest(options, callback);
+}
+
+(process.env.S3METADATA === 'mongodb' ? describe : describe.skip)('Veeam routes:', () => {
+    let scuba;
+
+    beforeEach(done => {
+        scuba = new MockScuba();
+        scuba.start();
+        setTimeout(done, 500);
+    });
+
+    afterEach(() => {
+        scuba.stop();
+    });
+
+    describe('veeam PUT routes:', () => {
+        before(done => {
+            bucketUtil = new BucketUtility(
+                'default', { signatureVersion: 'v4' });
+            s3 = bucketUtil.s3;
+            s3.createBucket({ Bucket: TEST_BUCKET }).promise()
```

Comment:
shouldn't we use await and stop using callbacks?

Reply:
Yes, changing the bucket util is likely a long work that I would prefer doing separately as we use it in many test files...

Reply:
sorry, i missed the rename of the file...